docs/pages/lighting-setups.rst

```diff
@@ -0,0 +1,197 @@
+Working with Lights
+###################
+
+:summary: This tutorial demonstrates the creation and manipulation of LightSetups in Habitat-sim.
+
+.. contents::
+    :class: m-block m-default
+
+A **LightSetup** consists of a set of LightInfo structures defining the common configuration
+of a set of point lights used to render objects in a scene. Once defined and registered,
+a LightSetup can be assigned to any subset of objects in the scene, including the scene asset itself.
+
+Each **LightInfo** structure in a LightSetup defines the `color`, `position`, and `LightPositionModel` of a single point light source.
+The LightPositionModel defines the coordinate frame of the light and can be set to any of:
+
+.. code:: python
+
+    habitat_sim.gfx.LightPositionModel.CAMERA
+    habitat_sim.gfx.LightPositionModel.GLOBAL
+    habitat_sim.gfx.LightPositionModel.OBJECT
+
+Each LightSetup is registered in the simulator via a unique key. Two default lighting setups are pre-defined:
+
+
+- **habitat_sim.gfx.DEFAULT_LIGHTING_KEY**
+
+  8 white, GLOBAL lights at scene bounding box corners.
+
+- **habitat_sim.gfx.NO_LIGHT_KEY**
+
+  0 lights (for Flat shaded assets).
+
+
+Additional custom setups can be created and registered via the **simulator.set_light_setup** function:
+
+.. code:: python
+
+    sim.set_light_setup(new_light_setup, "my_custom_lighting_key")
+
+Any existing LightSetup can be queried with **simulator.get_light_setup**:
+
+.. code:: python
+
+    custom_light_setup = sim.set_light_setup(n"my_custom_lighting_key")
+
+
+`Working with Lights Example:`_
+===============================
+
+The example code below demonstrates the usage of default and custom LightSetups and is runnable via:
+
+.. code:: shell-session
+
+    $ python examples/tutorials/lighting_tutorial.py
+
+First, we import modules we will need, define some convenience functions, and initialize the Simulator, Agent, and objects.
+
+.. include:: ../../examples/tutorials/lighting_tutorial.py
+    :code: py
+    :start-after: # [setup]
+    :end-before: # [/setup]
```

Comment:
It gets better! This is awesome! I wish we could similarly refer to cells in an ipynb. 

@mosra -- wish list :-).

Reply:
Since I have the whole stack under my control, nothing is impossible here (*evil laughter* :smiling_imp:).

I imagine Jupyter has some Python API for accessing notebook contents, so all that's needed to be done is hooking that up. I only have a "drive by" experience with Jupyter, can someone point me to APIs that could enable something like this?

```rst
.. jupyter:: bla.ipynb
    :cell: setup
```

Reply:
I also don't have tons of experience here. Maybe this? https://nbsphinx.readthedocs.io/en/0.6.0/index.html

Reply:
Oh, thanks! While I'm not running on top of Sphinx, I'm sure I could take this apart and extract valuable parts. Goes to my TODO list :)